skbio/metadata/_metadata.py

```diff
@@ -0,0 +1,1337 @@
+    dataframes.
+
+    ``Metadata`` objects have various methods to access, filter, and merge
+    data. A dataframe can be retrieved from the ``Metadata`` object for further
+    data manipulation using the pandas API. Individual ``MetadataColumn``
+    objects can be retrieved to gain access to APIs applicable to a single
+    metadata column.
+
+    Missing values may be encoded in one of the following schemes:
+
+    'blank'
+      The default, which treats `None`/`NaN` as the only valid missing values.
+
+    'no-missing'
+      Indicates there are no missing values in a column, any `None`/`NaN`
+      values should be considered an error. If a scheme other than 'blank' is
+      used by default, this scheme can be provided to preserve strings as
+      categorical terms.
+
+    'INSDC:missing'
+      The INSDC vocabulary for missing values. The current implementation
+      supports only lower-case terms which match exactly:
+      'not applicable', 'missing', 'not provided', 'not collected', and
+      'restricted access'.
+
+    Parameters
+    ----------
+    dataframe : pandas.DataFrame
+        Dataframe containing metadata. The dataframe's index defines the IDs,
+        and the index name (``Index.name``) must match one of the required ID
+        headers described in the metadata file format spec. Each column in the
+        dataframe defines a metadata column, and the metadata column's type
+        (i.e. *categorical* or *numeric*) is determined based on the column's
+        dtype. If a column has ``dtype=object``, it may contain strings or
+        pandas missing values (e.g. ``np.nan``, ``None``). Columns matching
+        this requirement are assumed to be *categorical*. If a column in the
+        dataframe has ``dtype=float`` or ``dtype=int``, it may contain floating
+        point numbers or integers, as well as pandas missing values
+        (e.g. ``np.nan``). Columns matching this requirement are assumed to be
+        *numeric*. Regardless of column type (categorical vs numeric), the
+        dataframe stored within the ``Metadata`` object will have any missing
+        values normalized to ``np.nan``. Columns with ``dtype=int`` will be
+        cast to ``dtype=float``. To obtain a dataframe from the ``Metadata``
+        object containing these normalized data types and values, use
+        ``Metadata.to_dataframe()``.
+    column_missing_schemes : dict, optional
+        Describe the metadata column handling for missing values described
+        in the dataframe. This is a dict mapping column names (str) to
+        missing-value schemes (str).  Valid values are 'blank',
+        'no-missing', and 'INSDC:missing'. Column names may be omitted.
+    default_missing_scheme : str, optional
+        The missing scheme to use when none has been provided in the file
+        or in `column_missing_schemes`.
+
+    """
+
+    @classmethod
+    def load(cls, filepath, column_types=None, column_missing_schemes=None,
```

Comment:
Adopting support for skbio's registry should be done, but I think we need to maintain this API for backward compatibility